libdnf/comps/comps.cpp

```diff
@@ -17,52 +30,38 @@ void Comps::load_installed() {
 }
 
 
-void Comps::load_from_file(const std::string & path) {
-    // xml parser context
-    xmlParserCtxtPtr xml_ctx = xmlNewParserCtxt();
-
-    if (xml_ctx == nullptr) {
-        throw std::bad_alloc();
-        // "Failed to allocate xml parser context"
-    }
-
-    // resulting xml document tree
-    // parse the file, activating the DTD validation option
-    // xml_doc = xmlCtxtReadFile(xml_ctx, path.c_str(), nullptr, XML_PARSE_DTDVALID);
-    xmlDocPtr xml_doc = xmlCtxtReadFile(xml_ctx, path.c_str(), nullptr, 0);
-
-    xmlNode * root_node = xmlDocGetRootElement(xml_doc);
-
-    // TODO(dmach): assert name == comps
+void Comps::load_from_file(const std::string & path, Repo * repo) {
```

Comment:
Since you're using pool instead of parsing XML directly,
it would be good to modify the method to only load
the comps content into the pool and then follow the best practices
from rpm Package and Sack:
- load everything in the pool first
- don't store anything in the objects directly, don't create any objects and keep them in the sack (== use pool exclusively)
- Sack would create objects on fly and return weak pointers the same way rpm Package is handled